jupyter_server/serverapp.py

```diff
@@ -1487,31 +1490,96 @@ def init_server_extension_config(self):
                 self.config.ServerApp.jpserver_extensions.update({modulename: enabled})
                 self.jpserver_extensions.update({modulename: enabled})
 
-    def init_server_extensions(self):
+        # Discover ExtensionApp's to load their config that might affect ServerApp.
+        self.enabled_extensions = {}
+        for module_name, enabled in sorted(self.jpserver_extensions.items()):
+            if enabled:
+                # Search for server extensions by loading each extension module
+                # and looking for the `_jupyter_server_extension_paths` function.
+                try:
+                    mod, metadata_list = _get_server_extension_metadata(module_name)
+                    for metadata in metadata_list:
+                        # Check if this server extension is a ExtensionApp object.
+                        extapp = metadata.get('app', None)
+                        extloc = metadata.get('module', None)
+                        # Load the extension module.
+                        if extapp and extloc:
```

Comment:
I read this logic as "if an ExtensionApp, use new initialization path, if not, use old one". It is unclear to me if we should initialize *all* `ExtensionApp`s this way, or only the one being initialized as the main app?

Reply:
I think all `ExtensionApp`s should be initialized the same way by the server... currently that's by creating and instance of the app and calling `link_to_serverapp`. 

I can see that it's a bit weird that I'm checking `extloc` here too. My intention was that `extloc` should always be found in `metadata`. If it's not found, the extension should not be initialized and a warning thrown below.